source/dubregistry/registry.d

```diff
@@ -431,11 +431,21 @@ class DubRegistry {
 
 	void updatePackages()
 	{
+		import std.random : randomShuffle;
+
 		logDiagnostic("Triggering package update...");
 		// update stat distributions before score packages
 		m_statDistributions = m_db.getStatDistributions();
-		foreach (packname; this.availablePackages)
-			triggerPackageUpdate(packname);
+
+		// shuffle the package list to ensure a uniform probability of each
+		// package getting chosen over time, regardless of how long it takes
+		// to process the whole list and whether the process gets restarted
+		// in-between
+		auto allpacks = this.availablePackages.array;
+		randomShuffle(allpacks);
+		foreach (packname; allpacks)
+			if (!m_updateQueue.isPending(packname))
```

Comment:
shouldn't this be handled by triggerPackageUpdate?

Reply:
`triggerPackageUpdate` is also used for the immediate update trigger, so that would then require further changes.